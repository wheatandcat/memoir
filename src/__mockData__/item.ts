import { ItemQuery } from 'queries/api/index';
type Item = NonNullable<ItemQuery['item']>;

export const item = (option?: Partial<Item>): Item => ({
  id: '1',
  title: '買い物',
  categoryID: 9,
  date: '2021-01-01T00:00:00+09:00',
  like: false,
  dislike: false,
  createdAt: '2021-01-01T00:00:00+09:00',
  updatedAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const items = (): Item[] => [
  item(),
  item({
    id: '2',
    title: '本を読む',
    categoryID: 2,
    date: '',
    dislike: false,
    like: false,
  }),
];
