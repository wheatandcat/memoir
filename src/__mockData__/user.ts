import { User } from 'queries/api/index';

export const user = (option?: Partial<User>): User => ({
  id: '1',
  displayName: 'test',
  image: '',
  updatedAt: '2021-01-01T00:00:00+09:00',
  createdAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const users = (): User[] => [
  {
    id: 'test1',
    displayName: 'suzuki1',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test2',
    displayName: 'suzuki2',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test3',
    displayName: 'suzuki4',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test4',
    displayName: 'suzuki4',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test5',
    displayName: 'suzuki5',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
];
