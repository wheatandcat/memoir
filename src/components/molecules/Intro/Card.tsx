import React, { memo } from 'react';
import {
  StyleSheet,
  ImageBackground,
  ImageSourcePropType,
  TouchableWithoutFeedback,
} from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import Image from 'components/atoms/Image';
import theme from 'config/theme';

export type Props = {
  source: ImageSourcePropType;
  text: string;
  onNext: () => void;
};

const Card: React.FC<Props> = (props) => {
  return (
    <ImageBackground
      source={require('../../../img/common/frame.png')}
      resizeMode="cover"
      style={styles.image}
    >
      <View style={styles.inner}>
        <View>
          <Image
            source={props.source}
            width={95}
            height={95}
            resizeMode="contain"
          />
        </View>
        <View mt={4}>
          <Text
            textAlign="center"
            variants="large"
            lineHeight={50}
            fontFamily="NotoSansJP-Bold"
          >
            {props.text}
          </Text>
        </View>
        <View mt={4}>
          <TouchableWithoutFeedback onPress={props.onNext}>
            <View style={styles.nextButton}>
              <Text fontWeight="bold" textAlign="center">
                次へ
              </Text>
            </View>
          </TouchableWithoutFeedback>
        </View>
      </View>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  image: {
    flex: 1,
    justifyContent: 'center',
    position: 'relative',
  },
  inner: {
    alignItems: 'center',
    height: '50%',
  },
  nextButton: {
    width: 180,
    backgroundColor: theme().color.background.main,
    paddingVertical: theme().space(3),
  },
});

export default memo(Card);
