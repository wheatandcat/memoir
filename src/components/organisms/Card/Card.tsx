import React, { memo } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  useWindowDimensions,
  ViewStyle,
} from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import Category from 'components/atoms/Category';
import theme from 'config/theme';
import Image from 'components/atoms/Image';
import { Item } from 'queries/api/index';
import setting from 'components/atoms/Category/setting';
import { categoryBorderStyle } from 'lib/category';

type User = {
  id: string;
  name: string;
};

export type Props = {
  title: Item['title'];
  categoryID: Item['categoryID'];
  user?: User;
  onPress: () => void;
};

const Card: React.FC<Props> = (props) => {
  const windowWidth = useWindowDimensions().width;

  const titleStyle: ViewStyle = {
    width: windowWidth - 120,
  };
  if (props.user) {
    titleStyle.height = 60;
    titleStyle.justifyContent = 'flex-end';
  }

  const rootStyle = { height: 75 };
  if (props.user) {
    rootStyle.height = 95;
  }

  const category = setting().icon.find(
    (v) => v.id === props.categoryID
  )?.category;
  const categoryStyle: ViewStyle[] = [
    rootStyle,
    styles.root,
    categoryBorderStyle(category || 0),
  ];

  return (
    <TouchableOpacity onPress={props.onPress}>
      <View style={categoryStyle}>
        <View mx={2}>
          <Category categoryID={props.categoryID} />
        </View>
        <View>
          <View style={[styles.title, titleStyle]}>
            <Text numberOfLines={2} ellipsizeMode="tail" lineHeight={25}>
              {props.title}
            </Text>
          </View>
          {!!props.user?.id && (
            <View style={styles.user}>
              <Image
                source={require('../../../img/icon/account.png')}
                width={20}
                height={20}
              />
              <View pl={2}>
                <Text variants="small">{props.user?.name}</Text>
              </View>
            </View>
          )}
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default memo(Card);

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.light,
    flexDirection: 'row',
    alignItems: 'center',
    borderLeftWidth: 5,
  },
  title: {
    paddingLeft: theme().space(0),
  },
  user: {
    paddingVertical: theme().space(2),
    flexDirection: 'row',
    alignItems: 'center',
  },
});
