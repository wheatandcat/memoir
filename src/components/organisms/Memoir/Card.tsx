import React, { memo, useEffect, useState } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  useWindowDimensions,
  ViewStyle,
} from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import Category from 'components/atoms/Category';
import theme from 'config/theme';
import { Item } from 'queries/api/index';
import UserImage from 'components/molecules/User/Image';

type User = {
  id: string;
  displayName: string;
  image: string;
};

export type Props = {
  title: Item['title'];
  categoryID: Item['categoryID'];
  user: User;
  onPress: () => void;
  onLoadEnd?: () => void;
};

const Card: React.FC<Props> = (props) => {
  const [endUserImage, setEndUserImage] = useState(false);
  const [endCategoryImage, setEndCategoryImage] = useState(false);
  const windowWidth = useWindowDimensions().width;

  useEffect(() => {
    if (endCategoryImage && endUserImage) {
      props.onLoadEnd?.();
    }
  }, [endCategoryImage, endUserImage, props]);

  const titleStyle: ViewStyle = {
    width: windowWidth - 130,
  };

  return (
    <TouchableOpacity onPress={props.onPress}>
      <View style={styles.root}>
        <View>
          <Category
            categoryID={props.categoryID}
            onLoadEnd={() => setEndCategoryImage(true)}
          />
        </View>
        <View pl={3}>
          <View style={[styles.title, titleStyle]}>
            <Text numberOfLines={2} ellipsizeMode="tail" lineHeight={25}>
              {props.title}
            </Text>
          </View>
          <View style={styles.user}>
            <UserImage
              image={props.user.image}
              size={20}
              onLoadEnd={() => setEndUserImage(true)}
            />
            <View pl={2}>
              <Text variants="small">{props.user?.displayName}</Text>
            </View>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default memo(Card);

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: theme().space(2),
  },
  title: {},
  user: {
    paddingVertical: theme().space(2),
    flexDirection: 'row',
    alignItems: 'center',
  },
});
