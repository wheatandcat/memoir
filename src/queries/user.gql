query User {
  user {
    id
    displayName
  }
}
