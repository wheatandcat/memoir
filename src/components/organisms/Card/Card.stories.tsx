import React from 'react';
import { mockFn } from 'storyBookUtils/index';
import Card from './Card';

export default {
  title: 'organisms/Card',
};

export const _Card = () => (
  <Card title="本を読んだ" categoryID={1} onPress={mockFn('onPress')} />
);
