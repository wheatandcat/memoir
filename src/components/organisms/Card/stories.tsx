import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { mockFn } from 'storyBookUtils/index';
import Card from './Card';

storiesOf('organisms/Card', module).add('Card', () => (
  <Card onPress={mockFn('onPress')} />
));
