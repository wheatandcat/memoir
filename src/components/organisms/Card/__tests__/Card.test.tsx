import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { item } from '__mockData__/item';
import { user } from '__mockData__/user';
import Card, { Props } from '../';

const propsData = (): Props => ({
  title: item().title,
  categoryID: item().categoryID,
  user: {
    id: user().id,
    name: 'name',
  },
  onPress: jest.fn(),
});

describe('components/organisms/Card/Card.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Card {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
