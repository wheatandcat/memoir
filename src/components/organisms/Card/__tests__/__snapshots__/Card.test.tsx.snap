// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/Card/Card.tsx 正常にrenderすること 1`] = `
<TouchableOpacity
  onPress={[MockFunction]}
>
  <View
    style={
      Array [
        Object {
          "height": 95,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderLeftWidth": 5,
          "flexDirection": "row",
        },
        Object {
          "borderLeftColor": "#FFE600",
        },
      ]
    }
  >
    <View
      mx={2}
    >
      <Memo(Category)
        categoryID={9}
      />
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "paddingLeft": 2,
            },
            Object {
              "height": 60,
              "justifyContent": "flex-end",
              "width": 630,
            },
          ]
        }
      >
        <Text
          ellipsizeMode="tail"
          fontFamily="RobotoCondensed-Bold"
          lineHeight={25}
          numberOfLines={2}
          variants="body"
        >
          買い物
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingVertical": 8,
          }
        }
      >
        <Image
          height={20}
          source={null}
          width={20}
        />
        <View
          pl={2}
        >
          <Text
            fontFamily="RobotoCondensed-Bold"
            variants="small"
          >
            name
          </Text>
        </View>
      </View>
    </View>
  </View>
</TouchableOpacity>
`;
